### DISCUSSION TITLE
Setting Login Mode on Windows Machine

### DISCUSSION THREAD

**Discussion by andynatx** [VladislavAntonyuk/MauiSamples#26] [Q&A] [answered]
When initiating the user login on the Windows machine how would I specify a redirect rather than a popup login? Could MsalProviderOptions.LoginMode do this?

**Comment by VladislavAntonyuk** [comment, marked as answer]
I have recently updated the sample, which should open browser instead of Popup.
See https://github.com/VladislavAntonyuk/MauiSamples/blob/main/Auth/AuthServices/BaseAuthService.cs#L19

You also need to redirect to localhost instead of msal.

**Comment by andynatx** [comment]
Thanks for that update. I actually tried adding .WithUseEmbeddedWebView(false), but I did not update the redirect.  One clarification, must I update it in the client config? Currently its set to the msal.

![image](https://user-images.githubusercontent.com/107771460/175783486-b60f0e8f-f6e2-43e1-8708-bd22f37c4fa7.png)
![image](https://user-images.githubusercontent.com/107771460/175783488-c29bbd31-978a-4a81-b55f-78b4ff0f9e76.png)
![image](https://user-images.githubusercontent.com/107771460/175783497-1af5f0fa-3184-42e2-83f2-26e531f06c69.png)

**Reply by VladislavAntonyuk** [reply]
please also take a look at PublicClientApplicationBuilder: https://github.com/VladislavAntonyuk/MauiSamples/blob/main/Auth/AuthServices/DependencyExtensions.cs#L14

**Comment by andynatx** [comment]
Got it working.  Its now opening in the browser.  Is there a flow that will open a login screen inside the application window? Not a pop-up and not a browser page with the dialog embedded.

**Reply by VladislavAntonyuk** [reply]
I think you can implement it (like open webpage in web view control), but it is not secure and may be forbidden by some identity providers

**Reply by rkaderli** [reply]
With Android it did work well for me, Thanks :-)

Then I tried B2C on Windows  and I tried to set WithUseEmbeddedWebView to true ... 

![image](https://user-images.githubusercontent.com/43808577/183098588-b37fb872-b3ff-4e53-98f1-bfcbbbb27684.png)

But it failed with unhandled Exception

![image](https://user-images.githubusercontent.com/43808577/183098561-d7ceaeeb-22b1-4179-954f-facc5b85bf6b.png)

Using external Chrome (my default) it does open the Browser but then is blocked (Even after allowing Popup etc). Any idea what else it could be?

**Reply by VladislavAntonyuk** [reply]
It’s hard to say from the screenshot. If you attach a small repro or send the exception I can try to help

**Reply by rkaderli** [reply]
I'm using a clone from Your repos ... So does Windows (WithUseEmbeddedWebView on true and false) work for You?

Regarding exception details, I could not attach a Debugger, possibly for security reason allow debug beeing disabled?

**Reply by VladislavAntonyuk** [reply]
yes, it works. just make sure you have set up a redirect to localhost in Azure Portal and in the application.
Did you follow the guide on how to setup the app and Azure: https://vladislavantonyuk.azurewebsites.net/articles/Azure-Active-Directory-authentication-in-.NET-MAUI
